[BUG] Invalid column name 'additional_duration_ms'

### Component

Full Dashboard

### Performance Monitor Version

2.0.0

### SQL Server Version

SQL Server 2022 Developer Edition

### Windows Version

Windows Server 2022 Standard

### Describe the Bug

After installing Perfmon and Dashboard 2.0 I saw this error in logfile 'PerformanceMonitor_20260226.log' under AppData\Local\PerformanceMonitor\Logs\

"[2026-02-26 08:33:49.107] [ERROR] Error loading query store regressions: Invalid column name 'additional_duration_ms'.
Invalid column name 'baseline_exec_count'.
Invalid column name 'recent_exec_count'.
Invalid column name 'baseline_plan_count'.
Invalid column name 'recent_plan_count'.
Invalid column name 'additional_duration_ms'."


### Steps to Reproduce

Installing PerformanceMonitor 2.0.0

### Expected Behavior

Expected no errors in logfile

### Actual Behavior

Errors in logfile

### Error Messages / Log Output

```text
"[2026-02-26 08:33:49.107] [ERROR] Error loading query store regressions: Invalid column name 'additional_duration_ms'.
Invalid column name 'baseline_exec_count'.
Invalid column name 'recent_exec_count'.
Invalid column name 'baseline_plan_count'.
Invalid column name 'recent_plan_count'.
Invalid column name 'additional_duration_ms'."
```

### Screenshots

_No response_

### Additional Context

All agents running OK
Monitoring local SQL instance.

--- a/Dashboard/CollectionLogWindow.xaml
+++ b/Dashboard/CollectionLogWindow.xaml
@@ -8,11 +8,19 @@
         Background="{DynamicResource BackgroundBrush}">
     <Window.Resources>
         <ContextMenu x:Key="DataGridContextMenu">
-            <MenuItem Header="Copy Cell" Click="CopyCell_Click"/>
-            <MenuItem Header="Copy Row" Click="CopyRow_Click"/>
-            <MenuItem Header="Copy All Rows" Click="CopyAllRows_Click"/>
+            <MenuItem Header="Copy Cell" Click="CopyCell_Click">
+                <MenuItem.Icon><TextBlock Text="&#x1F4CB;"/></MenuItem.Icon>
+            </MenuItem>
+            <MenuItem Header="Copy Row" Click="CopyRow_Click">
+                <MenuItem.Icon><TextBlock Text="&#x1F4C4;"/></MenuItem.Icon>
+            </MenuItem>
+            <MenuItem Header="Copy All Rows" Click="CopyAllRows_Click">
+                <MenuItem.Icon><TextBlock Text="&#x1F4D1;"/></MenuItem.Icon>
+            </MenuItem>
             <Separator/>
-            <MenuItem Header="Export to CSV..." Click="ExportToCsv_Click"/>
+            <MenuItem Header="Export to CSV..." Click="ExportToCsv_Click">
+                <MenuItem.Icon><TextBlock Text="&#x1F4CA;"/></MenuItem.Icon>
+            </MenuItem>
         </ContextMenu>
     </Window.Resources>
     <Grid>

--- a/Dashboard/CollectorScheduleWindow.xaml
+++ b/Dashboard/CollectorScheduleWindow.xaml
@@ -8,11 +8,19 @@
         Background="{DynamicResource BackgroundBrush}">
     <Window.Resources>
         <ContextMenu x:Key="DataGridContextMenu">
-            <MenuItem Header="Copy Cell" Click="CopyCell_Click"/>
-            <MenuItem Header="Copy Row" Click="CopyRow_Click"/>
-            <MenuItem Header="Copy All Rows" Click="CopyAllRows_Click"/>
+            <MenuItem Header="Copy Cell" Click="CopyCell_Click">
+                <MenuItem.Icon><TextBlock Text="&#x1F4CB;"/></MenuItem.Icon>
+            </MenuItem>
+            <MenuItem Header="Copy Row" Click="CopyRow_Click">
+                <MenuItem.Icon><TextBlock Text="&#x1F4C4;"/></MenuItem.Icon>
+            </MenuItem>
+            <MenuItem Header="Copy All Rows" Click="CopyAllRows_Click">
+                <MenuItem.Icon><TextBlock Text="&#x1F4D1;"/></MenuItem.Icon>
+            </MenuItem>
             <Separator/>
-            <MenuItem Header="Export to CSV..." Click="ExportToCsv_Click"/>
+            <MenuItem Header="Export to CSV..." Click="ExportToCsv_Click">
+                <MenuItem.Icon><TextBlock Text="&#x1F4CA;"/></MenuItem.Icon>
+            </MenuItem>
         </ContextMenu>
     </Window.Resources>
     <Grid Margin="16">

--- a/Dashboard/Controls/AlertsHistoryContent.xaml
+++ b/Dashboard/Controls/AlertsHistoryContent.xaml
@@ -13,11 +13,19 @@
 
             <!-- Context Menu for DataGrid Copy/Export -->
             <ContextMenu x:Key="DataGridContextMenu">
-                <MenuItem Header="Copy Cell" Click="CopyCell_Click"/>
-                <MenuItem Header="Copy Row" Click="CopyRow_Click"/>
-                <MenuItem Header="Copy All Rows" Click="CopyAllRows_Click"/>
+                <MenuItem Header="Copy Cell" Click="CopyCell_Click">
+                    <MenuItem.Icon><TextBlock Text="&#x1F4CB;"/></MenuItem.Icon>
+                </MenuItem>
+                <MenuItem Header="Copy Row" Click="CopyRow_Click">
+                    <MenuItem.Icon><TextBlock Text="&#x1F4C4;"/></MenuItem.Icon>
+                </MenuItem>
+                <MenuItem Header="Copy All Rows" Click="CopyAllRows_Click">
+                    <MenuItem.Icon><TextBlock Text="&#x1F4D1;"/></MenuItem.Icon>
+                </MenuItem>
                 <Separator/>
-                <MenuItem Header="Export to CSV..." Click="ExportToCsv_Click"/>
+                <MenuItem Header="Export to CSV..." Click="ExportToCsv_Click">
+                    <MenuItem.Icon><TextBlock Text="&#x1F4CA;"/></MenuItem.Icon>
+                </MenuItem>
             </ContextMenu>
 
             <Style x:Key="DefaultRowStyle" TargetType="DataGridRow">

--- a/Dashboard/MainWindow.xaml
+++ b/Dashboard/MainWindow.xaml
@@ -128,13 +128,23 @@
                     </ListView.ItemTemplate>
                     <ListView.ContextMenu>
                         <ContextMenu Opened="ServerContextMenu_Opened">
-                            <MenuItem Header="Open in New Tab" Click="OpenServerTab_Click"/>
-                            <MenuItem Header="Check Connection" Click="CheckConnection_Click"/>
-                            <MenuItem Header="Edit Server..." Click="EditServer_Click"/>
+                            <MenuItem Header="Open in New Tab" Click="OpenServerTab_Click">
+                                <MenuItem.Icon><TextBlock Text="&#x1F5C2;"/></MenuItem.Icon>
+                            </MenuItem>
+                            <MenuItem Header="Check Connection" Click="CheckConnection_Click">
+                                <MenuItem.Icon><TextBlock Text="&#x2705;"/></MenuItem.Icon>
+                            </MenuItem>
+                            <MenuItem Header="Edit Server..." Click="EditServer_Click">
+                                <MenuItem.Icon><TextBlock Text="&#x270F;"/></MenuItem.Icon>
+                            </MenuItem>
                             <Separator/>
-                            <MenuItem x:Name="ToggleFavoriteMenuItem" Header="Set as Favorite" Click="ToggleFavorite_Click"/>
+                            <MenuItem x:Name="ToggleFavoriteMenuItem" Header="Set as Favorite" Click="ToggleFavorite_Click">
+                                <MenuItem.Icon><TextBlock Text="&#x2B50;"/></MenuItem.Icon>
+                            </MenuItem>
                             <Separator/>
-                            <MenuItem Header="Remove Server" Click="RemoveServer_Click"/>
+                            <MenuItem Header="Remove Server" Click="RemoveServer_Click">
+                                <MenuItem.Icon><TextBlock Text="&#x1F5D1;"/></MenuItem.Icon>
+                            </MenuItem>
                         </ContextMenu>
                     </ListView.ContextMenu>
                 </ListView>

--- a/Dashboard/ManageServersWindow.xaml
+++ b/Dashboard/ManageServersWindow.xaml
@@ -9,11 +9,19 @@
     <Window.Resources>
         <BooleanToVisibilityConverter x:Key="BoolToVisibilityConverter"/>
         <ContextMenu x:Key="DataGridContextMenu">
-            <MenuItem Header="Copy Cell" Click="CopyCell_Click"/>
-            <MenuItem Header="Copy Row" Click="CopyRow_Click"/>
-            <MenuItem Header="Copy All Rows" Click="CopyAllRows_Click"/>
+            <MenuItem Header="Copy Cell" Click="CopyCell_Click">
+                <MenuItem.Icon><TextBlock Text="&#x1F4CB;"/></MenuItem.Icon>
+            </MenuItem>
+            <MenuItem Header="Copy Row" Click="CopyRow_Click">
+                <MenuItem.Icon><TextBlock Text="&#x1F4C4;"/></MenuItem.Icon>
+            </MenuItem>
+            <MenuItem Header="Copy All Rows" Click="CopyAllRows_Click">
+                <MenuItem.Icon><TextBlock Text="&#x1F4D1;"/></MenuItem.Icon>
+            </MenuItem>
             <Separator/>
-            <MenuItem Header="Export to CSV..." Click="ExportToCsv_Click"/>
+            <MenuItem Header="Export to CSV..." Click="ExportToCsv_Click">
+                <MenuItem.Icon><TextBlock Text="&#x1F4CA;"/></MenuItem.Icon>
+            </MenuItem>
         </ContextMenu>
     </Window.Resources>
     <Grid Margin="16">

--- a/Lite/Controls/AlertsHistoryTab.xaml
+++ b/Lite/Controls/AlertsHistoryTab.xaml
@@ -12,11 +12,19 @@
             </ResourceDictionary.MergedDictionaries>
 
             <ContextMenu x:Key="DataGridContextMenu">
-                <MenuItem Header="Copy Cell" Click="CopyCell_Click"/>
-                <MenuItem Header="Copy Row" Click="CopyRow_Click"/>
-                <MenuItem Header="Copy All Rows" Click="CopyAllRows_Click"/>
+                <MenuItem Header="Copy Cell" Click="CopyCell_Click">
+                    <MenuItem.Icon><TextBlock Text="&#x1F4CB;"/></MenuItem.Icon>
+                </MenuItem>
+                <MenuItem Header="Copy Row" Click="CopyRow_Click">
+                    <MenuItem.Icon><TextBlock Text="&#x1F4C4;"/></MenuItem.Icon>
+                </MenuItem>
+                <MenuItem Header="Copy All Rows" Click="CopyAllRows_Click">
+                    <MenuItem.Icon><TextBlock Text="&#x1F4D1;"/></MenuItem.Icon>
+                </MenuItem>
                 <Separator/>
-                <MenuItem Header="Export to CSV..." Click="ExportToCsv_Click"/>
+                <MenuItem Header="Export to CSV..." Click="ExportToCsv_Click">
+                    <MenuItem.Icon><TextBlock Text="&#x1F4CA;"/></MenuItem.Icon>
+                </MenuItem>
             </ContextMenu>
 
             <Style x:Key="DefaultRowStyle" TargetType="DataGridRow">

--- a/Lite/Controls/ServerTab.xaml
+++ b/Lite/Controls/ServerTab.xaml
@@ -11,15 +11,29 @@
 
             <!-- Context Menu for DataGrid Copy -->
             <ContextMenu x:Key="DataGridContextMenu">
-                <MenuItem Header="Copy Cell" Click="CopyCell_Click"/>
-                <MenuItem Header="Copy Row" Click="CopyRow_Click"/>
-                <MenuItem Header="Copy All Rows" Click="CopyAllRows_Click"/>
-                <MenuItem Header="Copy Repro Script" Click="CopyReproScript_Click"/>
+                <MenuItem Header="Copy Cell" Click="CopyCell_Click">
+                    <MenuItem.Icon><TextBlock Text="&#x1F4CB;"/></MenuItem.Icon>
+                </MenuItem>
+                <MenuItem Header="Copy Row" Click="CopyRow_Click">
+                    <MenuItem.Icon><TextBlock Text="&#x1F4C4;"/></MenuItem.Icon>
+                </MenuItem>
+                <MenuItem Header="Copy All Rows" Click="CopyAllRows_Click">
+                    <MenuItem.Icon><TextBlock Text="&#x1F4D1;"/></MenuItem.Icon>
+                </MenuItem>
+                <MenuItem Header="Copy Repro Script" Click="CopyReproScript_Click">
+                    <MenuItem.Icon><TextBlock Text="&#x1F4DD;"/></MenuItem.Icon>
+                </MenuItem>
                 <Separator/>
-                <MenuItem Header="Export to CSV..." Click="ExportToCsv_Click"/>
+                <MenuItem Header="Export to CSV..." Click="ExportToCsv_Click">
+                    <MenuItem.Icon><TextBlock Text="&#x1F4CA;"/></MenuItem.Icon>
+                </MenuItem>
                 <Separator/>
-                <MenuItem Header="View Plan" Click="ViewEstimatedPlan_Click"/>
-                <MenuItem Header="Get Actual Plan" Click="GetActualPlan_Click"/>
+                <MenuItem Header="View Plan" Click="ViewEstimatedPlan_Click">
+                    <MenuItem.Icon><TextBlock Text="&#x1F50D;"/></MenuItem.Icon>
+                </MenuItem>
+                <MenuItem Header="Get Actual Plan" Click="GetActualPlan_Click">
+                    <MenuItem.Icon><TextBlock Text="&#x25B6;"/></MenuItem.Icon>
+                </MenuItem>
             </ContextMenu>
 
             <Style x:Key="GridRowStyle" TargetType="DataGridRow">

--- a/Lite/MainWindow.xaml
+++ b/Lite/MainWindow.xaml
@@ -69,13 +69,23 @@
                                     Background="{DynamicResource BackgroundLightBrush}">
                                 <Border.ContextMenu>
                                     <ContextMenu>
-                                        <MenuItem Header="Connect" Click="ServerContextMenu_Connect_Click"/>
-                                        <MenuItem Header="Disconnect" Click="ServerContextMenu_Disconnect_Click"/>
+                                        <MenuItem Header="Connect" Click="ServerContextMenu_Connect_Click">
+                                            <MenuItem.Icon><TextBlock Text="&#x1F517;"/></MenuItem.Icon>
+                                        </MenuItem>
+                                        <MenuItem Header="Disconnect" Click="ServerContextMenu_Disconnect_Click">
+                                            <MenuItem.Icon><TextBlock Text="&#x274C;"/></MenuItem.Icon>
+                                        </MenuItem>
                                         <Separator/>
-                                        <MenuItem Header="Toggle Favorite" Click="ServerContextMenu_ToggleFavorite_Click"/>
+                                        <MenuItem Header="Toggle Favorite" Click="ServerContextMenu_ToggleFavorite_Click">
+                                            <MenuItem.Icon><TextBlock Text="&#x2B50;"/></MenuItem.Icon>
+                                        </MenuItem>
                                         <Separator/>
-                                        <MenuItem Header="Edit..." Click="ServerContextMenu_Edit_Click"/>
-                                        <MenuItem Header="Remove" Click="ServerContextMenu_Remove_Click"/>
+                                        <MenuItem Header="Edit..." Click="ServerContextMenu_Edit_Click">
+                                            <MenuItem.Icon><TextBlock Text="&#x270F;"/></MenuItem.Icon>
+                                        </MenuItem>
+                                        <MenuItem Header="Remove" Click="ServerContextMenu_Remove_Click">
+                                            <MenuItem.Icon><TextBlock Text="&#x1F5D1;"/></MenuItem.Icon>
+                                        </MenuItem>
                                     </ContextMenu>
                                 </Border.ContextMenu>
                                 <Grid>

--- a/Lite/Windows/CollectionLogWindow.xaml
+++ b/Lite/Windows/CollectionLogWindow.xaml
@@ -11,11 +11,19 @@
                 <ResourceDictionary Source="/Themes/DarkTheme.xaml"/>
             </ResourceDictionary.MergedDictionaries>
             <ContextMenu x:Key="DataGridContextMenu">
-                <MenuItem Header="Copy Cell" Click="CopyCell_Click"/>
-                <MenuItem Header="Copy Row" Click="CopyRow_Click"/>
-                <MenuItem Header="Copy All Rows" Click="CopyAllRows_Click"/>
+                <MenuItem Header="Copy Cell" Click="CopyCell_Click">
+                    <MenuItem.Icon><TextBlock Text="&#x1F4CB;"/></MenuItem.Icon>
+                </MenuItem>
+                <MenuItem Header="Copy Row" Click="CopyRow_Click">
+                    <MenuItem.Icon><TextBlock Text="&#x1F4C4;"/></MenuItem.Icon>
+                </MenuItem>
+                <MenuItem Header="Copy All Rows" Click="CopyAllRows_Click">
+                    <MenuItem.Icon><TextBlock Text="&#x1F4D1;"/></MenuItem.Icon>
+                </MenuItem>
                 <Separator/>
-                <MenuItem Header="Export to CSV..." Click="ExportToCsv_Click"/>
+                <MenuItem Header="Export to CSV..." Click="ExportToCsv_Click">
+                    <MenuItem.Icon><TextBlock Text="&#x1F4CA;"/></MenuItem.Icon>
+                </MenuItem>
             </ContextMenu>
         </ResourceDictionary>
     </Window.Resources>

--- a/Lite/Windows/ManageServersWindow.xaml
+++ b/Lite/Windows/ManageServersWindow.xaml
@@ -12,11 +12,19 @@
                 <ResourceDictionary Source="/Themes/DarkTheme.xaml"/>
             </ResourceDictionary.MergedDictionaries>
             <ContextMenu x:Key="DataGridContextMenu">
-                <MenuItem Header="Copy Cell" Click="CopyCell_Click"/>
-                <MenuItem Header="Copy Row" Click="CopyRow_Click"/>
-                <MenuItem Header="Copy All Rows" Click="CopyAllRows_Click"/>
+                <MenuItem Header="Copy Cell" Click="CopyCell_Click">
+                    <MenuItem.Icon><TextBlock Text="&#x1F4CB;"/></MenuItem.Icon>
+                </MenuItem>
+                <MenuItem Header="Copy Row" Click="CopyRow_Click">
+                    <MenuItem.Icon><TextBlock Text="&#x1F4C4;"/></MenuItem.Icon>
+                </MenuItem>
+                <MenuItem Header="Copy All Rows" Click="CopyAllRows_Click">
+                    <MenuItem.Icon><TextBlock Text="&#x1F4D1;"/></MenuItem.Icon>
+                </MenuItem>
                 <Separator/>
-                <MenuItem Header="Export to CSV..." Click="ExportToCsv_Click"/>
+                <MenuItem Header="Export to CSV..." Click="ExportToCsv_Click">
+                    <MenuItem.Icon><TextBlock Text="&#x1F4CA;"/></MenuItem.Icon>
+                </MenuItem>
             </ContextMenu>
         </ResourceDictionary>
     </Window.Resources>

--- a/Lite/Windows/ProcedureHistoryWindow.xaml
+++ b/Lite/Windows/ProcedureHistoryWindow.xaml
@@ -12,11 +12,19 @@
                 <ResourceDictionary Source="/Themes/DarkTheme.xaml"/>
             </ResourceDictionary.MergedDictionaries>
             <ContextMenu x:Key="DataGridContextMenu">
-                <MenuItem Header="Copy Cell" Click="CopyCell_Click"/>
-                <MenuItem Header="Copy Row" Click="CopyRow_Click"/>
-                <MenuItem Header="Copy All Rows" Click="CopyAllRows_Click"/>
+                <MenuItem Header="Copy Cell" Click="CopyCell_Click">
+                    <MenuItem.Icon><TextBlock Text="&#x1F4CB;"/></MenuItem.Icon>
+                </MenuItem>
+                <MenuItem Header="Copy Row" Click="CopyRow_Click">
+                    <MenuItem.Icon><TextBlock Text="&#x1F4C4;"/></MenuItem.Icon>
+                </MenuItem>
+                <MenuItem Header="Copy All Rows" Click="CopyAllRows_Click">
+                    <MenuItem.Icon><TextBlock Text="&#x1F4D1;"/></MenuItem.Icon>
+                </MenuItem>
                 <Separator/>
-                <MenuItem Header="Export to CSV..." Click="ExportToCsv_Click"/>
+                <MenuItem Header="Export to CSV..." Click="ExportToCsv_Click">
+                    <MenuItem.Icon><TextBlock Text="&#x1F4CA;"/></MenuItem.Icon>
+                </MenuItem>
             </ContextMenu>
         </ResourceDictionary>
     </Window.Resources>

--- a/Lite/Windows/QueryStatsHistoryWindow.xaml
+++ b/Lite/Windows/QueryStatsHistoryWindow.xaml
@@ -12,11 +12,19 @@
                 <ResourceDictionary Source="/Themes/DarkTheme.xaml"/>
             </ResourceDictionary.MergedDictionaries>
             <ContextMenu x:Key="DataGridContextMenu">
-                <MenuItem Header="Copy Cell" Click="CopyCell_Click"/>
-                <MenuItem Header="Copy Row" Click="CopyRow_Click"/>
-                <MenuItem Header="Copy All Rows" Click="CopyAllRows_Click"/>
+                <MenuItem Header="Copy Cell" Click="CopyCell_Click">
+                    <MenuItem.Icon><TextBlock Text="&#x1F4CB;"/></MenuItem.Icon>
+                </MenuItem>
+                <MenuItem Header="Copy Row" Click="CopyRow_Click">
+                    <MenuItem.Icon><TextBlock Text="&#x1F4C4;"/></MenuItem.Icon>
+                </MenuItem>
+                <MenuItem Header="Copy All Rows" Click="CopyAllRows_Click">
+                    <MenuItem.Icon><TextBlock Text="&#x1F4D1;"/></MenuItem.Icon>
+                </MenuItem>
                 <Separator/>
-                <MenuItem Header="Export to CSV..." Click="ExportToCsv_Click"/>
+                <MenuItem Header="Export to CSV..." Click="ExportToCsv_Click">
+                    <MenuItem.Icon><TextBlock Text="&#x1F4CA;"/></MenuItem.Icon>
+                </MenuItem>
             </ContextMenu>
         </ResourceDictionary>
     </Window.Resources>

--- a/Lite/Windows/QueryStoreHistoryWindow.xaml
+++ b/Lite/Windows/QueryStoreHistoryWindow.xaml
@@ -12,11 +12,19 @@
                 <ResourceDictionary Source="/Themes/DarkTheme.xaml"/>
             </ResourceDictionary.MergedDictionaries>
             <ContextMenu x:Key="DataGridContextMenu">
-                <MenuItem Header="Copy Cell" Click="CopyCell_Click"/>
-                <MenuItem Header="Copy Row" Click="CopyRow_Click"/>
-                <MenuItem Header="Copy All Rows" Click="CopyAllRows_Click"/>
+                <MenuItem Header="Copy Cell" Click="CopyCell_Click">
+                    <MenuItem.Icon><TextBlock Text="&#x1F4CB;"/></MenuItem.Icon>
+                </MenuItem>
+                <MenuItem Header="Copy Row" Click="CopyRow_Click">
+                    <MenuItem.Icon><TextBlock Text="&#x1F4C4;"/></MenuItem.Icon>
+                </MenuItem>
+                <MenuItem Header="Copy All Rows" Click="CopyAllRows_Click">
+                    <MenuItem.Icon><TextBlock Text="&#x1F4D1;"/></MenuItem.Icon>
+                </MenuItem>
                 <Separator/>
-                <MenuItem Header="Export to CSV..." Click="ExportToCsv_Click"/>
+                <MenuItem Header="Export to CSV..." Click="ExportToCsv_Click">
+                    <MenuItem.Icon><TextBlock Text="&#x1F4CA;"/></MenuItem.Icon>
+                </MenuItem>
             </ContextMenu>
         </ResourceDictionary>
     </Window.Resources>

--- a/install/47_create_reporting_views.sql
+++ b/install/47_create_reporting_views.sql
@@ -1848,7 +1848,9 @@ RETURN
             query_id = qsd.query_id,
             avg_duration_ms = AVG(qsd.avg_duration / 1000.0),
             avg_cpu_time_ms = AVG(qsd.avg_cpu_time / 1000.0),
-            avg_logical_io_reads = AVG(qsd.avg_logical_io_reads)
+            avg_logical_io_reads = AVG(qsd.avg_logical_io_reads),
+            exec_count = SUM(qsd.count_executions),
+            plan_count = COUNT(DISTINCT qsd.plan_id)
         FROM collect.query_store_data AS qsd
         WHERE qsd.server_last_execution_time < @start_date
         GROUP BY
@@ -1864,6 +1866,8 @@ RETURN
             avg_duration_ms = AVG(qsd.avg_duration / 1000.0),
             avg_cpu_time_ms = AVG(qsd.avg_cpu_time / 1000.0),
             avg_logical_io_reads = AVG(qsd.avg_logical_io_reads),
+            exec_count = SUM(qsd.count_executions),
+            plan_count = COUNT(DISTINCT qsd.plan_id),
             last_execution_time = MAX(qsd.server_last_execution_time)
         FROM collect.query_store_data AS qsd
         WHERE qsd.server_last_execution_time >= @start_date
@@ -1887,6 +1891,12 @@ RETURN
         recent_reads = r.avg_logical_io_reads,
         io_regression_percent =
             CONVERT(decimal(10,2), (r.avg_logical_io_reads - b.avg_logical_io_reads) * 100.0 / NULLIF(b.avg_logical_io_reads, 0)),
+        additional_duration_ms =
+            CONVERT(decimal(18,2), (r.avg_duration_ms - b.avg_duration_ms) * r.exec_count),
+        baseline_exec_count = b.exec_count,
+        recent_exec_count = r.exec_count,
+        baseline_plan_count = b.plan_count,
+        recent_plan_count = r.plan_count,
         severity =
             CASE
                 WHEN (r.avg_duration_ms - b.avg_duration_ms) * 100.0 / NULLIF(b.avg_duration_ms, 0) > 100 THEN N'CRITICAL'
@@ -1894,7 +1904,7 @@ RETURN
                 WHEN (r.avg_duration_ms - b.avg_duration_ms) * 100.0 / NULLIF(b.avg_duration_ms, 0) > 25 THEN N'MEDIUM'
                 ELSE N'LOW'
             END,
-        query_text_sample = CONVERT(nvarchar(500), r.query_text_sample),
+        query_text_sample = r.query_text_sample,
         last_execution_time = r.last_execution_time
     FROM recent_performance AS r
     JOIN baseline_performance AS b
@@ -1902,7 +1912,7 @@ RETURN
       AND b.query_id = r.query_id
     WHERE (r.avg_cpu_time_ms - b.avg_cpu_time_ms) * 100.0 / NULLIF(b.avg_cpu_time_ms, 0) > 25
     ORDER BY
-        (r.avg_cpu_time_ms - b.avg_cpu_time_ms) * 100.0 / NULLIF(b.avg_cpu_time_ms, 0) DESC
+        additional_duration_ms DESC
 );
 GO
 

